ros_buildfarm/templates/release/rpm/mock_config.cfg.em

```diff
@@ -10,7 +18,12 @@ config_opts['use_bootstrap'] = False
 config_opts['chroot_setup_cmd'] += ' python3-rpm-macros'
 
 # Install weak dependencies to get group members
-config_opts[f'dnf_builddep_opts'] = config_opts.get(f'dnf_builddep_opts', []) + ['--setopt=install_weak_deps=True']
+package_manager = _expanded('package_manager', config_opts)
+config_opts[f'{package_manager}_builddep_opts'] = config_opts.get(f'{package_manager}_builddep_opts', []) + ['--setopt=install_weak_deps=True']
+
+# Deal with dnf -> dnf4 transition - see rpm-software-management/mock#1496
+if package_manager == 'dnf':
+    config_opts[f'dnf4_builddep_opts'] = config_opts.get(f'dnf4_builddep_opts', []) + ['--setopt=install_weak_deps=True']
```

Comment:
nit: Since we're setting it already a few lines above, we can avoid code duplication

```suggestion
    config_opts[f'dnf4_builddep_opts'] = config_opts.get(f'dnf4_builddep_opts', []) + config_opts.get(f'dnf_builddep_opts', [])
```

Reply:
Compromise: since `dnf_builddep_opts` and `dnf4_builddep_opts` could theoretically be different before this code executes, I pulled the list of options we want to add into a variable and appended that to each of the lists, so there is only one list to maintain.